cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,75 @@
 /// <reference types='cypress' />
-
+import { generateUser } from '../support/generate';
 describe('Student Registration page', () => {
   before(() => {
-
+    cy.visit('https://demoqa.com/automation-practice-form');
   });
 
-  it('', () => {
+  const { email, firstName, lastName, address } = generateUser();
+  const number = 1234567891;
+
+  it('should enter vaild data to register', () => {
+    cy.get('#firstName').type(firstName);
+
+    cy.get('#lastName').type(lastName);
+
+    cy.get('#userEmail').type(email);
+
+    cy
+      .get('#genterWrapper > .col-md-9 > :nth-child(1) > .custom-control-label')
+      .click();
+
+    cy.get('#userNumber').type(number);
+
+    cy.get('.subjects-auto-complete__value-container').type('Eng');
+
+    cy.get('#react-select-2-option-0').click();
```

Comment:
It looks like you're trying to select an option from a dropdown menu. Cypress has a plugin called 'cypress-select-plugin' which provides a custom command to make it easier to select options from dropdown menus. Consider using this plugin to make your code cleaner and more readable.